### PR TITLE
add emio

### PR DESCRIPTION
**Pull request by Viatorus**
Specify library name and version:  **emio/0.3.0**

em{io} is a safe and fast high-level and low-level character input/output library for bare-metal and RTOS based embedded systems with a very small binary footprint.

I am the author of this package. It is currently in beta (that's why 0.3.0). I would like to add it to the conan center.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Viatorus** [comment]
Could anybody please help me here?

I don't understand what the problem in v1 is (and how to solve it) and how can I test apple-clang without having a Mac for v2?

File: recipes/emio/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "*.hpp", src=os.path.join(self.source_folder, "include"),
+             dst=os.path.join(self.package_folder, "include"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "emio"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "emio"
+        self.cpp_info.names["cmake_find_package"] = "emio"
+        self.cpp_info.names["cmake_find_package_multi"] = "emio"
```
Comment (memsharded) [inline recipes/emio/all/conanfile.py:68]: Not necessary to support legacy generators in new recipes, you can drop this too.
You might need to set ``cpp_info.set_property("cmake_target_name", ...`` (the filename not necessary if it matches the package name).

File: recipes/emio/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class EmioConan(ConanFile):
+    name = "emio"
+    description = "A character input/output library for embedded systems."
+    topics = ("format", "scan", "header-only")
+    homepage = "https://github.com/viatorus/emio"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
```
Comment (memsharded) [inline recipes/emio/all/conanfile.py:20]:
```suggestion
    package_type = "header-library"
    no_copy_source = True
```

To be more ready for Conan 2.0

File: recipes/emio/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES CXX)
+
+find_package(emio REQUIRED)
+
+add_executable(test_package test_package.cpp)
+target_link_libraries(test_package PRIVATE emio::emio)
```
Comment (memsharded) [inline recipes/emio/all/test_package/CMakeLists.txt:7]: This doesn't match the above target names!

File: recipes/emio/all/conanfile.py (hunk 1 above)
Comment (memsharded) [reply recipes/emio/all/conanfile.py:68]: Apparently, based on the below usage, this wouldn't be necessary at all, can be dropped completely.

**Review by memsharded — CHANGES_REQUESTED** [review]

File: recipes/emio/all/conanfile.py (hunk 2 above)
Comment (Viatorus) [reply recipes/emio/all/conanfile.py:21]: Changed as suggested.

File: recipes/emio/all/conanfile.py (hunk 1 above)
Comment (Viatorus) [reply recipes/emio/all/conanfile.py]: Deleted package info.

File: recipes/emio/all/test_v1_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.8)
+project(test_package LANGUAGES CXX)
+
```
[the remark this answers is not in the document]
Comment (Viatorus) [reply recipes/emio/all/test_v1_package/CMakeLists.txt]: Deleted folder `test_v1_package`.

File: recipes/emio/all/test_package/CMakeLists.txt (hunk 3 above)
Comment (Viatorus) [reply recipes/emio/all/test_package/CMakeLists.txt:7]: Hmm, I don't get it. The target name is everwhere the same: `test_package`

Comment (memsharded) [reply recipes/emio/all/test_package/CMakeLists.txt:7]: No prob, it was related to the code that was removed, that apparently tried to define a different target name like only ``emio`` and not ``emio::emio``. but this is good.

**Review by memsharded — APPROVED** [review]
Thanks for doing the review, good job, approving.
Thanks for your contribution!

**Review by RubenRBS — APPROVED** [review]
Thanks a lot for your contribution :)

**Comment by Viatorus** [comment]
Thank you guys!

**Pull request merged**